crates/uv/src/commands/project/run.rs

```diff
@@ -56,6 +58,27 @@ pub(crate) async fn run(
         warn_user_once!("`uv run` is experimental and may change without warning.");
     }
 
+    // These cases seem quite complex because it changes the current package — let's just
+    // ban it entirely for now
```

Comment:
Alternatively, `uv run -r` could imply `--isolated` (which might make sense but is more limiting?)

Reply:
Am I correct that the _code_ would work, and here, you just mean that semantically `-r foo/bar/pyproject.toml` _should_ use `foo/bar` as the project?

Reply:
Yeah the code should work as-is, it's just weird since it may _imply_ a change of project.

Reply:
Like.. you should use `--with ./foo/bar` instead, right?